mapper/orm-coordination-outbox-polling/src/main/java/org/hibernate/search/mapper/orm/coordination/outboxpolling/event/impl/OutboxPollingEventProcessor.java

```diff
@@ -275,9 +277,21 @@ public CompletableFuture<?> work() {
 						// (complete() will be called, re-scheduling the polling for later)
 						return;
 					}
-					List<OutboxEvent> events = eventFinder.get().findOutboxEvents( session, batchSize );
-					if ( events.isEmpty() ) {
-						// Nothing to do, try again later (complete() will be called, re-scheduling the polling for later)
+
+					List<OutboxEvent> events;
+					try {
+						events = eventFinder.get().findOutboxEvents( session, batchSize );
+						if ( events.isEmpty() ) {
+							// Nothing to do, try again later (complete() will be called, re-scheduling the polling for later)
+							return;
+						}
+					}
+					catch (OptimisticLockException lockException) {
+						// Don't be fooled by the exception type, this is actually a *pessimistic* lock failure.
```

Comment:
That's ... quite intuitive 😄 
Is it CockroachDB that returns false info, or it's just the way ORM behaves?
BTW, should we catch the pessimistic one as well?

Reply:
>  Is it CockroachDB that returns false info, or it's just the way ORM behaves?

Just the way ORM behaves.

>  BTW, should we catch the pessimistic one as well?

I'll do that, but I've never seen ORM throw one.